maven/flatpak-maven-generator.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python3
+
+__license__ = 'MIT'
+
+import argparse
+import hashlib
+import json
+import logging
+import re
+import shutil
+import sys
+import tempfile
+from typing import List, Dict, Optional
+import urllib.request
+import xml.etree.ElementTree as ET
+
+parser = argparse.ArgumentParser()
+parser.add_argument('packages', nargs='*')
+parser.add_argument('--output', '-o',
+                    help='Specify output file name', default="maven-sources.json")
+parser.add_argument('--repo', '-r', action="append")
+parser.add_argument('--verbose', '-v', action='store_true')
+
+def assembleUri(repo: str, groupId: str, artifactId: str, version: str, classifier: Optional[str], extension: str) -> str:
+    groupId = groupId.replace(".", "/")
+    if(classifier is not None):
+        return f'{repo}{groupId}/{artifactId}/{version}/{artifactId}-{version}-{classifier}.{extension}'
+    else:
+        return f'{repo}{groupId}/{artifactId}/{version}/{artifactId}-{version}.{extension}'
+
+def getFileHash(file) -> str:
+    file.seek(0)
+    byteData = file.read() # read entire file as bytes
+    return hashlib.sha256(byteData).hexdigest()
+
+def parsePomDeps(parsed_pom) -> List[Dict[str, str]]:
+    ns = {'POM': 'http://maven.apache.org/POM/4.0.0'}
+
+    result = []
+
+    parent = parsed_pom.find("POM:parent", ns)
+    if parent is not None:
+        groupId = parent.find("POM:groupId", ns).text
+        artifactId = parent.find("POM:artifactId", ns).text
+        version = parent.find("POM:version", ns).text
+
+        result.append({
+            "groupId": groupId,
+            "artifactId": artifactId,
+            "version": version
+        })
+
+    deps = parsed_pom.findall("POM:dependencies/POM:dependency", ns)
+    for dep in deps:
+        groupId = dep.find("POM:groupId", ns).text
+        artifactId = dep.find("POM:artifactId", ns).text
+        version = dep.find("POM:version", ns)
+
+        if(version is None):
+            continue
```

Comment:
Can you check out what's the matter with [com.esotericsoftware:kryo](https://repo1.maven.org/maven2/com/esotericsoftware/kryo/4.0.1/kryo-4.0.1.pom)? This check (although IMO really sensible) skips its dependency on `reflectasm` (gradle resolves the version to `1.11.3`).

Reply:
Ugh, I figured it out. `kryo`'s parent ([`kryo-parent`](https://repo1.maven.org/maven2/com/esotericsoftware/kryo-parent/4.0.1/kryo-parent-4.0.1.pom) somehow defines the version through a property in a `<dependencyManagement>` tag.

You know what, screw that dependency, I'll add it manually to the command.

Reply:
Are there cases where silently continuing here is desired? If not, maybe log a warning and instruct the user to add it to the dependencies manually.

Oh, and thank you a lot for your work! You're awesome!